### ISSUE TITLE
Using next-marker of a list call in other list calls

### ISSUE THREAD

**Issue by rohangarg** [Azure/azure-storage-java#252]
I have the following blobs in my container in an azure blob storage account : 
1. a/b/c
2. a/b/c1
3. a/c

I start with a list call with prefix a/b/ and with maxresults as 1. I get the object a/b/c and a continuation token. Now, the next call I make is with prefix a/ and with continuation token from the previous list call. The maxresults param is 2. The objects I get are : a/b/c1 and a/c. Is it ok to assume consistency in this behavior? i.e if I use the continuation token of a list call in another list call with a different prefix, the nature of listing will be similar to startAfterKey (w.r.t to the continuation token).

**Comment by mirobers** [comment]
The continuation token is a marker sufficient to continue the original operation successfully. This implies that your example will work the way you describe, as the marker must be somewhere between the first and second blobs.

However, I'd advise you not to take this dependency. Without blob 2, the first call might not return a continuation token at all. There are additional pitfalls when adding, removing, or changing the delimiter during continuation, or changing the include= filter.

**Comment by rohangarg** [comment]
@mirobers : Yes, the points you raised are valid. Although we do not currently change 'include=' parameter and keep the delimiter as null always, but still I would follow your advise to ensure correct and deterministic behavior.
But can this be taken as a feature request? (providing a startAfterKey option in listing) or if present already, any expected timelines for it?
This feature would immensely help/speed-up our listing use-case.
